bears/general/CasingBear.py

```diff
@@ -0,0 +1,137 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.naming_conventions import (
+    to_camelcase, to_pascalcase, to_snakecase)
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class CasingBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results,
+            casing: str,
+            language: str):
+        """
+        Checks whether all identifier names (variables, classes, objects)
+        follow a certain naming convention.
+
+        :param casing:   camelCasing or snake_casing or PascalCasing
+        :param language: The language of the file, which is used to determine
+                         the keywords to ignore.
+        """
+        casing_convention = {"camel": to_camelcase,
+                             "pascal": to_pascalcase,
+                             "snake": to_snakecase}
+
+        if casing not in casing_convention:
+            self.err("Invalid file-naming-convention provided: " + casing)
+            return
+
+        self.convertor = casing_convention[casing]
+        coalang = LanguageDefinition(language)
+
+        if "keywords" not in coalang or "special_chars" not in coalang:
+            self.warn("CasingBear did not run because coalang definition "
+                      "is not available for " + language)
+            return
+
+        self.keywords = coalang["keywords"]
+        self.delim = re.escape(str(coalang["special_chars"]) + " \t\n")
+
+        annotations = dependency_results[AnnotationBear.name][0].contents
+        string_results = annotations["strings"]
+        results = []
+        last_line = 1
+        last_column = 1
+        text = "".join(file)
+        for string_range in string_results:
+            results.append(SourceRange.from_values(
+                text,
+                start_line=last_line,
+                start_column=last_column,
+                end_line=string_range.start.line,
+                end_column=string_range.start.column))
+            last_line = string_range.end.line
+            last_column = string_range.end.column
+        results.append(SourceRange.from_values(
+            text,
+            start_line=last_line,
+            start_column=last_column,
+            end_line=len(file),
+            end_column=len(file[-1])))
+
+        changes = {}
+        for result in results:
+            l = result.start.line
```

Comment:
so its not a good practice to use single-letter variable names and especially just a lower-case L  since it looks too much like a 1